merkledag/node.go

```diff
@@ -76,6 +76,11 @@ func MakeLink(n *Node) (*Link, error) {
 	}, nil
 }
 
+// GetCachedNode returns the MDAG Node that was cached, or nil
+func (l *Link) GetCachedNode() *Node {
```

Comment:
Yeah, I think we can recoup the perf losses elsewhere, by adding an lru cache to the dagservice (or elsewhere in the stack).